src/CrudPanel.php

```diff
@@ -237,16 +239,91 @@ public function sort($items)
      *          App/Models/Address defined by a company() method on the user model and an address() method on the
      *          company model, the 'App/Models/Address' string will be returned.
      *
-     * @param $relationString String Relation string. A dot notation can be used to chain multiple relations.
+     * @param string $relationString Relation string. A dot notation can be used to chain multiple relations.
+     * @param int $length Optionally specify the number of relations to omit from the start of the relation string. If
+     *        the provided length is negative, then that many relations will be omitted from the end of the relation
+     *        string.
+     * @param Model $model Optionally specify a different model than the one in the crud object.
      *
-     * @return string relation model name
+     * @return string Relation model name.
      */
-    private function getRelationModel($relationString)
+    public function getRelationModel($relationString, $length = null, $model = null)
     {
-        $result = array_reduce(explode('.', $relationString), function ($obj, $method) {
+        $relationArray = explode('.', $relationString);
+
+        if (! isset($length)) {
+            $length = count($relationArray);
+        }
+
+        if (! isset($model)) {
+            $model = $this->model;
+        }
+
+        $result = array_reduce(array_splice($relationArray, 0, $length), function ($obj, $method) {
             return $obj->$method()->getRelated();
-        }, $this->model);
+        }, $model);
 
         return get_class($result);
     }
+
+    /**
+     * Get the given attribute from a model or models resulting from the specified relation string (eg: the list of streets from
+     * the many addresses of the company of a given user).
+     *
+     * @param Model $model Model (eg: user).
+     * @param string $relationString Model relation. Can be a string representing the name of a relation method in the given
+     *        Model or one from a different Model through multiple relations. A dot notation can be used to specify
+     *        multiple relations (eg: user.company.address).
+     * @param string $attribute The attribute from the relation model (eg: the street attribute from the address model).
+     *
+     * @return array An array containing a list of attributes from the resulting model.
+     */
+    public function getModelAttributeFromRelation($model, $relationString, $attribute)
```

Comment:
If this returns an array, that Attribute should be plural, shouldn't it? Unless I'm confused...

Reply:
Well... sort of. It's a single attribute, but the end result of the relation string may lead to a collection of objects. So I end up getting a single attribute, but from many objects.

Consider the following hypothetical example:

You have a Product model with a one-to-one relation to a ProductDetails model which, in turn has a many-to-many relation to a ProductCategory. The relation string to the latter would then be something like `details.category`.

You would then formulate the problem something along the lines of 'I want the value for the  X attribute from this relation starting from this model'

If the `$attribute` parameter is then, let's say, 'name', you would end up with an array of the values of the 'name' attribute from each of the resulting collection of ProductCategory models.

This was the reasoning behind it, and I hope it makes things a little more clearer, but I can change it if you think the plural makes more sense.

Reply:
`getValueOfModelAttributeFromRelations` and now we're in the realms of awful names.

Why are we getting objects? Are attributes objects or the values of the attributes or am I just even **more** confued now?

Reply:
Let's look at the tables for the previously mentioned example:

| Product | 
| --------- |
| id |
| name |

| ProductDetails | 
| ---------------- |
| id |
| product_id |
| price |
| description |

| DetailsCategories | 
| ------------------- |
| id |
| product_details_id |
| product_category_id |

| ProductCategories | 
| ---------------- |
| id |
| name |

To get the value of the price attribute from the ProductDetails model through the details relation of the Product model, you would call the method like this:
`getModelAttributeFromRelation($productModel, 'details', 'price')`, which will return an array with zero or one entry (one-to-one relation).

To get the value of the name attribute from the Categories model through the categories relation defined in the ProductDetails model, you would do it like this:
`getModelAttributeFromRelation($productModel, 'details.categories', 'name')`, which will return an array with zero or many entries (many-to-many relation).

Reply:
Ok, this sounds like it's probably actually a few tests of differing names; we can tease them out later.